### PR TITLE
GUVNOR-2273 : Replace "package name white list" with Dependency Filter and related UI improvements

### PR DESCRIPTION
**Pull request by Rikkola**
Scope is no longer editable in the dependency grid.
You can white list the packages in the listed jar or remove from the white list.
Transient dependencies are also listed.
Packages in the current project are always white listed, but everything else needs to be added.
Also removed some redundant code and split few classes into smaller units.

https://github.com/droolsjbpm/kie-docs/pull/32
https://github.com/droolsjbpm/guvnor/pull/219
https://github.com/droolsjbpm/kie-wb-common/pull/202
https://github.com/droolsjbpm/drools-wb/pull/108
https://github.com/droolsjbpm/jbpm-console-ng/pull/261
https://github.com/droolsjbpm/jbpm-form-modeler/pull/32

### CODE REVIEW

File: guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java]: 2016!

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java [hunk 2]
```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: 2016!

File: guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java (hunk 1 above)
Comment (Rikkola) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java]: It was 2015 when I wrote it :(

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+                             final CommentedOptionFactory commentedOptionFactory,
+                             final BackwardCompatibleUtil backward ) {
+        this.ioService = ioService;
+        this.pomService = pomService;
+        this.configurationService = configurationService;
+        this.commentedOptionFactory = commentedOptionFactory;
+        this.backward = backward;
+    }
+
+    public Package doNewPackage( final Package parentPackage,
```
Comment (manstis) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: "doNewPackage" isn't a really a nice method name; simply "newPackage" would probably be better (yes, I know this as copied and pasted... but it moves from protected to public)

**Comment by manstis** [comment]
Couple of small suggestions; otherwise +1 to merge.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java (hunk 3 above)
Comment (Rikkola) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: Do it! 
Good point. I'll change that.

**Comment by Rikkola** [comment]
Fixes in.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ProjectResourcePaths.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.guvnor.common.services.project.backend.server;
```
Comment (manstis) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ProjectResourcePaths.java]: Oops, sorry.. no license :(

Comment (psiroky) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ProjectResourcePaths.java]: Looks like we really should add checkstyle/enforcer config which would fail the build in case of missing (or malformed) license headers. @mbiarnes could you take a look at that when you have some time?

Comment (Rikkola) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ProjectResourcePaths.java]: I tried to setup a check like that years ago, but at that time the tooling was not ready. But it might work today, if not for malformed, then just for missing headers.

Comment (Rikkola) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ProjectResourcePaths.java]: Ok, fixed.

**Comment by manstis** [comment]
+1 to merge.

**Comment by porcelli** [comment]
https://github.com/droolsjbpm/guvnor/commit/4c5b75932c0d8c9765f9c7202268435901a937a3

**Comment by psiroky** [comment]
Looks like there is one other PR in the chain which we were waiting for https://github.com/droolsjbpm/kie-wb-common/pull/202. The build of kie-wb-common is currently broken because of that.

**Comment by manstis** [comment]
...and there's one for drools-wb (IIRC Rikkola was looking into failing
tests on Jenkins for kie-wb-common?)

**Comment by csadilek** [comment]
@manstis @psiroky there are 5 more PRs here and we're waiting for @Rikkola to update https://github.com/droolsjbpm/kie-wb-common/pull/202. @porcelli I guess you merged this too early. So best if I merge all the other ones, I guess and we'll fix the test issue later?

**Comment by manstis** [comment]
I'm happy with that - I have some work in progress (BPMSPL-252) that needs
some heavy refactoring following these PRs.. and would prefer to get done
sooner that later :-)

**Comment by porcelli** [comment]
my bad, as I maybe did it too early. BUT in the other handm my regular process to merge has been passed, I meant... it build local (mvn clean install), what happens is sorta new to me is that we're doing -T2C - and this is faling.

**Comment by csadilek** [comment]
@porcelli no worries. I will merge the others. This is somewhat of an edge cases with that many PRs involved. However, I did experience the kie-wb-common test failures on my local machine even without -T....so something is up there. We have to investigate that separately. /cc @manstis @psiroky @Rikkola

**Comment by Rikkola** [comment]
I'm looking at the test failure issues. I did manage to reproduce them once. Feels like they are random.
